handler/oauth2/flow_refresh.go

```diff
@@ -25,6 +25,8 @@ import (
 	"context"
 	"time"
 
+	"github.com/ory/fosite/util"
```

Comment:
`util` is a go package antipattern :)

> Avoid meaningless package names. Packages named util, common, or misc provide clients with no sense of what the package contains. This makes it harder for clients to use the package and makes it harder for maintainers to keep the package focused. Over time, they accumulate dependencies that can make compilation significantly and unnecessarily slower, especially in large programs. And since such package names are generic, they are more likely to collide with other packages imported by client code, forcing clients to invent names to distinguish them.  - *[Source](https://blog.golang.org/package-names)*

One possible place could be `internal` for this (as it will be hidden from the outside).

Reply:
Ah cool, TIL 🙌 

Although if I place it in `internal`, the `Transactional` interface won't be visible by a consumer, e.g. `Hydra`. I need that to be exportable.

Reply:
I think that would be ok, or is there a reason why it should be visible?

Reply:
The only reason I can think of is for unit testing 🤔 

Mmm...maybe it is not necessary?

Reply:
I think that should be explicit in unit tests - whereas here we have implicit checking (because both non-tx and tx would work). So I think it's ok to not expose it!